File: Sem5/MA/flutter/tasks_application/lib/model/domain/task.dart
import 'package:tasks_application/model/domain/priority.dart';
import 'package:tasks_application/model/domain/status.dart';

class Task {
  int id;
  final String domain;
  final String title;
  final String description;
  final DateTime createdTime;
  final Priority priority;
  final DateTime deadline;
  final Status status;
  final int progress;

  Task(
      this.id,
      this.domain,
      this.title,
      this.description,
      this.createdTime,
      this.priority,
      this.deadline,
      this.status,
      this.progress);
}

File: Sem5/MA/android/TasksApplication/app/src/main/java/com/example/tasksapplication/model/Task.kt
package com.example.tasksapplication.model

import androidx.room.Entity
import androidx.room.PrimaryKey
import com.google.gson.annotations.SerializedName

@Entity(tableName = "tasks")
data class Task(
    @PrimaryKey(autoGenerate = false)
    @SerializedName("id")
    val id: Int = 0,
    @SerializedName("domain")
    val domain: String,
    @SerializedName("title")
    val title: String,
    @SerializedName("description")
    val description: String,
    @SerializedName("createdTime")
    val createdTime: Long,
    @SerializedName("priority")
    val priority: Priority,
    @SerializedName("deadline")
    val deadline: Long,
    @SerializedName("status")
    val status: Status,
    @SerializedName("progress")
    val progress: Int
)
